components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2243,22 +2251,45 @@ private AssociatedRolesConfig getAssociatedRoles(String applicationId, Connectio
         String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantID);
         AssociatedRolesConfig associatedRolesConfig = new AssociatedRolesConfig();
         List<String> associatedRoleIds = new ArrayList<>();
-        try (NamedPreparedStatement preparedStatement = new NamedPreparedStatement(connection,
-                ApplicationMgtDBQueries.LOAD_ASSOCIATED_ROLES)) {
-            preparedStatement.setString(ApplicationMgtDBQueries.SQLPlaceholders.DB_SCHEMA_COLUMN_NAME_APP_ID,
-                    applicationId);
-            try (ResultSet resultSet = preparedStatement.executeQuery()) {
-                while (resultSet.next()) {
-                    associatedRoleIds.add(resultSet.getString(1));
+        String allowedAudience =
+                getSPPropertyValueByPropertyKey(applicationId, ALLOWED_ROLE_AUDIENCE_PROPERTY_NAME, tenantDomain);
+        if (RoleConstants.APPLICATION.equalsIgnoreCase(allowedAudience)) {
+            try (NamedPreparedStatement preparedStatement = new NamedPreparedStatement(connection,
+                    ApplicationMgtDBQueries.LOAD_ASSOCIATED_ROLES)) {
+                preparedStatement.setString(ApplicationMgtDBQueries.SQLPlaceholders.DB_SCHEMA_COLUMN_NAME_APP_ID,
+                        applicationId);
+                try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                    while (resultSet.next()) {
+                        associatedRoleIds.add(resultSet.getString(1));
+                    }
                 }
+
+                associatedRolesConfig.setRoles(buildAssociatedRolesWithRoleName(associatedRoleIds, tenantDomain));
+            } catch (SQLException | IdentityRoleManagementException e) {
+                throw new IdentityApplicationManagementException(
+                        "Error while retrieving associated roles for application ID: " + applicationId, e);
             }
-            associatedRolesConfig.setRoles(buildAssociatedRolesWithRoleName(associatedRoleIds, tenantDomain));
-        } catch (SQLException | IdentityRoleManagementException e) {
-            throw new IdentityApplicationManagementException(
-                    "Error while retrieving associated roles for application ID: " + applicationId, e);
+        } else if (RoleConstants.ORGANIZATION.equalsIgnoreCase(allowedAudience)) {
+            ApplicationManagementServiceComponentHolder holder = ApplicationManagementServiceComponentHolder.
+                    getInstance();
+            RoleManagementService roleManagementService = holder.getRoleManagementServiceV2();
+            try {
+                if (roleManagementService != null) {
+                    List<RoleBasicInfo> listOfAllRoles = roleManagementService.
+                            getRoles(RoleConstants.AUDIENCE + SPACE + RoleConstants.EQ + SPACE +
```

Comment:
Is there any max limit defined in role mgt service when request to get all roles matching to a filter?

Reply:
Seems this is fixed in the same PR. Hence ignore the comment